keystore/filesystem/filesystem_backup.go

```diff
@@ -82,57 +83,58 @@ func isPrivate(fname string) bool {
 	return true
 }
 
-func getIDFromFilename(fname string) []byte {
+func getIDFromFilename(fname string) (keystore.KeyPurpose, []byte) {
 	if isHistoricalFilename(fname) {
 		fname = filepath.Dir(fname)
 	}
 	if fname == PoisonKeyFilename {
-		return []byte(fname)
+		return keystore.PurposePoisonRecordKeyPair, []byte(fname)
 	}
 	if fname == getSymmetricKeyName(PoisonKeyFilename) {
-		return []byte(fname[:len(fname)-len("_sym")])
+		return keystore.PurposePoisonRecordSymmetricKey, []byte(fname[:len(fname)-len("_sym")])
 	}
 	fname = filepath.Base(fname)
 	if strings.HasSuffix(fname, ".old") {
 		fname = fname[:len(fname)-len(".old")]
 	}
 	if strings.HasSuffix(fname, "_hmac") {
-		return []byte(fname[:len(fname)-len("_hmac")])
+		return keystore.PurposeSearchHMAC, []byte(fname[:len(fname)-len("_hmac")])
 	}
 	if strings.HasSuffix(fname, "_server") {
-		return []byte(fname[:len(fname)-len("_server")])
+		return keystore.PurposeLegacy, []byte(fname[:len(fname)-len("_server")])
 	}
 	if strings.HasSuffix(fname, "_translator") {
-		return []byte(fname[:len(fname)-len("_translator")])
+		return keystore.PurposeLegacy, []byte(fname[:len(fname)-len("_translator")])
 	}
 	if strings.HasSuffix(fname, "_storage") {
-		return []byte(fname[:len(fname)-len("_storage")])
+		return keystore.PurposeStorageClientPrivateKey, []byte(fname[:len(fname)-len("_storage")])
 	}
 	if strings.HasSuffix(fname, "_storage_sym") {
-		return []byte(fname[:len(fname)-len("_storage_sym")])
+		return keystore.PurposeStorageClientSymmetricKey, []byte(fname[:len(fname)-len("_storage_sym")])
 	}
 	if strings.HasSuffix(fname, "_zone") {
-		return []byte(fname[:len(fname)-len("_zone")])
+		return keystore.PurposeStorageZonePrivateKey, []byte(fname[:len(fname)-len("_zone")])
 	}
 	if strings.HasSuffix(fname, "_zone_sym") {
-		return []byte(fname[:len(fname)-len("_zone_sym")])
+		return keystore.PurposeStorageZoneSymmetricKey, []byte(fname[:len(fname)-len("_zone_sym")])
 	}
 	if strings.HasSuffix(fname, "_sym") {
-		return []byte(fname[:len(fname)-len("_sym")])
+		//TODO: what it the sym key here?
```

Comment:
@Lagovas, I think this is question to you

Reply:
probably zone's sym key - https://github.com/cossacklabs/acra/blob/master/keystore/filesystem/server_keystore.go#L943